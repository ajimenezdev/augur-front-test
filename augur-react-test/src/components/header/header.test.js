import React from "react";
import renderer from "react-test-renderer";
import { fireEvent, render, waitForElement } from "@testing-library/react";
import { act } from "@testing-library/react-hooks";
import Header from "./index";

it("renders without crashing", () => {
  const component = renderer.create(
    <Header token="" onTokenChange={() => {}} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders token address passed", () => {
  act(() => {
    const { getByText } = render(
      <Header
        token="0x1985365e9f78359a9b6ad760e32412f4a445e862"
        onTokenChange={() => {}}
      />
    );

    expect(getByText(/0x1985365e9f78359a9b6ad760e32412f4a445e862/i));
  });
});

it("execute callback when button is clicked", async () => {
  await act(async () => {
    const mockToken = jest.fn();
    const { getByLabelText, container } = render(
      <Header
        token="0x1985365e9f78359a9b6ad760e32412f4a445e862"
        onTokenChange={mockToken}
      />
    );

    const buttonChange = getByLabelText("change-token");
    fireEvent.click(buttonChange);

    await waitForElement(() => getByLabelText("token"), {
      container
    });

    fireEvent.change(getByLabelText("token"), {
      target: { value: "0xc350de406f54271c0e025bf72855418d65f8bcbc" }
    });
    const buttonSave = getByLabelText("save-button");
    fireEvent.click(buttonSave);

    expect(mockToken.mock.calls.length).toBe(1);
  });
});
