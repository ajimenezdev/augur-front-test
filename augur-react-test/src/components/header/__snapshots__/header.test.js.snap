// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<header
  className="header"
>
  <span
    className="header-title"
  >
    Header
  </span>
  <div
    className="header-token"
  >
    <span>
      Token: 
      
    </span>
    <button
      aria-label="change-token"
      className="MuiButtonBase-root MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <span
          aria-hidden={true}
          className="material-icons MuiIcon-root header-editIcon"
        >
          edit
        </span>
      </span>
    </button>
  </div>
</header>
`;
